gtk4-macros/src/lib.rs

```diff
@@ -15,6 +15,49 @@ use proc_macro::TokenStream;
 use proc_macro_error::proc_macro_error;
 use syn::{parse_macro_input, DeriveInput};
 
+/// That macro includes and compiles blueprint file by path relative to project rood
+///
+/// It expected to run inside composite_template_derive, not by users
+#[cfg(feature = "blueprint")]
+#[proc_macro]
+pub fn include_blueprint(input: TokenStream) -> TokenStream {
+    use quote::quote;
+
+    let tokens: Vec<_> = input.into_iter().collect();
+
+    if tokens.len() != 1 {
+        panic!("File name not found");
```

Comment:
Not sure if the panics here are macros friendly. cc @GuillaumeGomez @jf2048

Reply:
[`pest_derive`](https://docs.rs/pest_derive/latest/src/pest_derive/lib.rs.html#322-325) calls [`derive_parser`](https://docs.rs/pest_generator/2.5.7/src/pest_generator/lib.rs.html#43-109) from `pest_generator` that panics

Reply:
No they're not. You're supposed to return an error so people have an idea where it crashed and why. If the crate used panics, then another crate is needed.

Reply:
Then don't use `anyhow` either. A proc macro is supposed to return a `TokenStream`, so if you encounter an error, you generate a `TokenStream` containing the right span and `compile_error!` with the correct message.

You can find an example [here](https://github.com/gtk-rs/gtk-rs-core/blob/master/glib-macros/src/clone.rs#L67).

Reply:
That macro is called inside [`gen_set_template`](https://github.com/gtk-rs/gtk4-rs/pull/1348/files#:~:text=%23crate_ident%3A%3Agtk4_macros%3A%3Ainclude_blueprint!(%23file).as_bytes()) and I can't get correct span inside it, so there's no difference between `abort_call_site` and `compile_error!`

![image](https://user-images.githubusercontent.com/63008755/229344155-6e718e03-5a9e-4e15-9b08-083a1989f99f.png)
![image](https://user-images.githubusercontent.com/63008755/229344217-fad5326e-0e09-49ba-81df-214ff862dea2.png)